### DISCUSSION TITLE
Saving a screenshot from the rendering window when and as often as I want

### DISCUSSION THREAD

**Discussion by ldv1** [pyvista/pyvista#3352] [Q&A] [answered]
Hi,

When plotting with the interactive rendering windows in VTK, I would like to save a screenshot when and as often as I want. Is there a way to achieve that ?
Thanks for the help.

**Comment by banesullivan** [comment, marked as answer]
There are actually a lot of ways you could implement this. With a basic `pyvista.Plotter`, I recommend using `add_key_event` with a callback to save the screenshot. Here is an example of how to press the S-key to save a screenshot (incrementing file name as well):

```py
import pyvista as pv

class ScreenshotHandler:
    def __init__(self, plotter, path_no_ext):
        self.plotter = plotter
        self.path_no_ext = path_no_ext
        self.number = 0
        
    def __call__(self, *args):
        path = f"{self.path_no_ext}_{self.number:03d}.png"
        self.plotter.screenshot(filename=path)
        self.number += 1


p = pv.Plotter(notebook=0)
handler = ScreenshotHandler(p, './foo')
p.add_key_event('s', handler)
p.add_text('Press `s` to save screensot')
p.add_mesh(pv.Wavelet(), show_edges=True)
p.show()
```

**Reply by ldv1** [reply]
This is exactly what I was looking for !
Big thanks !
